fastlane/swift/LaneFileProtocol.swift

```diff
@@ -17,20 +17,20 @@ public protocol LaneFileProtocol: class {
     func recordLaneDescriptions()
     func beforeAll(with lane: String)
     func afterAll(with lane: String)
-    func onError(currentLane: String, errorInfo: String)
+    func onError(currentLane: String, errorInfo: String, errorClass: String?, errorMessage: String?)
 }
 
 public extension LaneFileProtocol {
     var fastlaneVersion: String { return "" } // Defaults to "" because that means any is fine
     func beforeAll(with _: String) {} // No-op by default
     func afterAll(with _: String) {} // No-op by default
-    func onError(currentLane _: String, errorInfo _: String) {} // No-op by default
     func recordLaneDescriptions() {} // No-op by default
 }
 
 @objcMembers
 open class LaneFile: NSObject, LaneFileProtocol {
     private(set) static var fastfileInstance: LaneFile?
+    private static var onErrorCalled: [String: Bool] = [:]
```

Comment:
Just to understand better, why we use a `[String: Bool]`?
I notice that we associate only `true` and never `false`. Maybe a `Set<String>` or an `Array<String>` could be a more indicated structure?

```suggestion
    private static var onErrorCalled: Set<String> = []
```

Reply:
The approach for a dictionary is mostly wise in terms of optimization, as reading in a dictionary takes O(1) whereas in an array or set it takes O(N). Not that the performance will be an issue here, but it was just pure instinct.

Reply:
I totally agree with the optimization point; array isn't the best solution, so that I suggested `Set`.
Also `Set` has `O(1)` reading / writing complexity and since in the dictionary we only add `true` values for lanes, it could make sense to semplificate the structure and the future validation operations and checks. ;)

Reply:
You're completely right! I thought array and set shared same complexity, so your approach is much better than the dictionary for the reasons you mentioned. I'll address this once I get energy to pick the laptop in the weekend😂.